internal/tokens/fetcher.go

```diff
@@ -0,0 +1,311 @@
+				}
+			}
+		}
+
+		/******************************************************************************************
+		** Checking if the token is a Compound token. We can determined that if we got a valid
+		** response from the `underlying` RPC call.
+		** If so, we set the token type to `Compound` and we add the coin we got as response for
+		** this RPC call in the newToken UnderlyingTokensAddresses.
+		** We can also add the coins to the relatedTokensList, so we can fetch it's information
+		** later.
+		******************************************************************************************/
+		isCToken := rawCUnderlying != nil && rawCUnderlying[0].(common.Address) != common.Address{}
+		if isCToken {
+			newToken.Type = `Compound`
+			coin := rawCUnderlying[0].(common.Address)
+			if (coin != common.Address{}) {
+				relatedTokensList = append(relatedTokensList, coin)
+				newToken.UnderlyingTokensAddresses = append(newToken.UnderlyingTokensAddresses, coin)
+			}
+		}
+
+		/******************************************************************************************
+		** Checking if the token is a AAVE V1 token. We can determined that if we got a valid
+		** response from the `underlyingAssetAddress` RPC call.
+		** If so, we set the token type to `AAVE V1` and we add the coin we got as response for
+		** this RPC call in the newToken UnderlyingTokensAddresses.
+		** We can also add the coins to the relatedTokensList, so we can fetch it's information
+		** later.
+		******************************************************************************************/
+		isAV1Token := rawAV1Underlying != nil && rawAV1Underlying[0].(common.Address) != common.Address{}
+		if isAV1Token {
+			newToken.Type = `AAVE V1`
+			coin := rawAV1Underlying[0].(common.Address)
+			if (coin != common.Address{}) {
+				relatedTokensList = append(relatedTokensList, coin)
+				newToken.UnderlyingTokensAddresses = append(newToken.UnderlyingTokensAddresses, coin)
+			}
+		}
+
+		/******************************************************************************************
+		** Checking if the token is a AAVE V2 token. We can determined that if we got a valid
+		** response from the `UNDERLYING_ASSET_ADDRESS` RPC call.
+		** If so, we set the token type to `AAVE V2` and we add the coin we got as response for
+		** this RPC call in the newToken UnderlyingTokensAddresses.
+		** We can also add the coins to the relatedTokensList, so we can fetch it's information
+		** later.
+		******************************************************************************************/
+		isAV2Token := rawAV2Underlying != nil && rawAV2Underlying[0].(common.Address) != common.Address{}
+		if isAV2Token {
+			newToken.Type = `AAVE V2`
+			coin := rawAV2Underlying[0].(common.Address)
+			if (coin != common.Address{}) {
+				relatedTokensList = append(relatedTokensList, coin)
+				newToken.UnderlyingTokensAddresses = append(newToken.UnderlyingTokensAddresses, coin)
+			}
+		}
+
```

Comment:
do we have cases when the above scenarios happen simultaneously?

Reply:
but after giving another thought, checking the values all at once through the batch call seems much cleaner, dropping this comment